lang/cs/Org.Apache.REEF.Tests/Functional/IMRU/IMRUBrodcastReduceTestBase.cs

```diff
@@ -0,0 +1,312 @@
+        /// <typeparam name="TMapOutput"></typeparam>
+        /// <typeparam name="TResult"></typeparam>
+        /// <typeparam name="TPartitionType"></typeparam>
+        /// <param name="numberOfTasks"></param>
+        /// <param name="driverId"></param>
+        /// <returns></returns>
+        protected IConfiguration CreateGroupCommunicationConfiguration<TMapInput, TMapOutput, TResult, TPartitionType>(
+            int numberOfTasks,
+            string driverId)
+        {
+            return TangFactory.GetTang().NewConfigurationBuilder()
+                .BindStringNamedParam<GroupCommConfigurationOptions.DriverId>(driverId)
+                .BindStringNamedParam<GroupCommConfigurationOptions.MasterTaskId>(IMRUConstants.UpdateTaskName)
+                .BindStringNamedParam<GroupCommConfigurationOptions.GroupName>(IMRUConstants.CommunicationGroupName)
+                .BindIntNamedParam<GroupCommConfigurationOptions.FanOut>(IMRUConstants.TreeFanout.ToString(CultureInfo.InvariantCulture).ToString(CultureInfo.InvariantCulture))
+                .BindIntNamedParam<GroupCommConfigurationOptions.NumberOfTasks>(numberOfTasks.ToString(CultureInfo.InvariantCulture))
+                .BindImplementation(GenericType<IGroupCommDriver>.Class, GenericType<GroupCommDriver>.Class)
+                .Build();
+        }
+
+        /// <summary>
+        /// Create IMRU Job Definition with IMRU required configurations
+        /// </summary>
+        /// <param name="numberofMappers"></param>
+        /// <param name="chunkSize"></param>
+        /// <param name="numIterations"></param>
+        /// <param name="dim"></param>
+        /// <param name="mapperMemory"></param>
+        /// <param name="updateTaskMemory"></param>
+        /// <returns></returns>
+        protected IMRUJobDefinition CreateIMRUJobDefinitionBuilder(int numberofMappers,
+            int chunkSize,
+            int numIterations,
+            int dim,
+            int mapperMemory,
+            int updateTaskMemory)
+        {
+            var updateFunctionConfig =
+                TangFactory.GetTang().NewConfigurationBuilder(BuildUpdateFunctionConfig())
+                    .BindNamedParameter(typeof(BroadcastReduceConfiguration.NumberOfIterations),
+                        numIterations.ToString(CultureInfo.InvariantCulture))
+                    .BindNamedParameter(typeof(BroadcastReduceConfiguration.Dimensions),
+                        dim.ToString(CultureInfo.InvariantCulture))
+                    .BindNamedParameter(typeof(BroadcastReduceConfiguration.NumWorkers),
+                        numberofMappers.ToString(CultureInfo.InvariantCulture))
+                    .Build();
+
+            return new IMRUJobDefinitionBuilder()
+                .SetMapFunctionConfiguration(BuildMapperFunctionConfig())
+                .SetUpdateFunctionConfiguration(updateFunctionConfig)
+                .SetMapInputCodecConfiguration(BuildUpdateFunctionCodecsConfig())
```

Comment:
Should this be `.SetMapInputCodecConfiguration(BuildMapInputCodecConfig())`? Otherwise `BuildMapInputCodecConfig()` is unused